Explain how to test pull requests and branches with our vagrant box (#510)
PrestaShop/docs: src/content/1.7/contribute/contribute-by-testing-pull-requests.md

---
title: Contribute by testing Pull Requests and branches
weight: 50
menuTitle: Test Pull Requests and branches
---

> # Why testing Pull Requests

Reviewer: ```suggestion
# Why test Pull Requests
```

Tests determine the quality of a software. This process involves evaluating information that is related to a shop.

> To help on testing, we provide a [Vagrant](http://www.vagrantup.com/) configuration designed for testing branches and pull requests.

Reviewer: ```suggestion
To make testing easier, we provide a [Vagrant](http://www.vagrantup.com/) configuration designed for testing branches and pull requests.
```

Maybe explain what vagrant is and why you can use it.

## What is installed

- Debian 10 Buster
- Git
- Multiple PHP versions (7.1, 7.2, 7.3, 7.4)
- MariaDB 10
- NodeJS

## Requires

First of all you need two tools to run this project:

- [Vagrant](https://www.vagrantup.com/downloads.html)
- [VirtualBox](https://www.virtualbox.org/wiki/Downloads)

If you're using Windows, we recommend you to download [Git bash](https://git-scm.com/downloads) and execute all commands under this shell.

## Why Vagrant

We use Vagrant because of its flexibility and portability.
With it, you can control every aspect of the machine, and help us by having a similar environment to reproduce issues.

## Getting started

> Go to [https://github.com/PrestaShop/vagrant](https://github.com/PrestaShop/vagrant) and follow instructions.

Reviewer: This probably needs to be replaced with a step by step guide with screenshots.

---
title: Contribute by testing Pull Requests and branches
weight: 50
menuTitle: Test Pull Requests and branches
---

# Why test Pull Requests

Tests determine the quality of a software. This process involves evaluating information that is related to a shop.

To make testing easier, we provide a [Vagrant](http://www.vagrantup.com/) configuration designed for testing branches and pull requests.

## What is installed

- Debian 10 Buster
- Git

> - Multiple PHP versions (7.1, 7.2, 7.3, 7.4)

Reviewer 2: ```suggestion
- Apache
- Multiple PHP versions (7.1, 7.2, 7.3, 7.4)
```

- MariaDB 10
- NodeJS

## Requires

First of all you need two tools to run this project:

- [Vagrant](https://www.vagrantup.com/downloads.html)
- [VirtualBox](https://www.virtualbox.org/wiki/Downloads)

If you're using Windows, we recommend you to download [Git bash](https://git-scm.com/downloads) and execute all commands under this shell.

## Why Vagrant

We use Vagrant because of its flexibility and portability.
With it, you can control every aspect of the machine, and help us by having a similar environment to reproduce issues.

## Getting started

The process is pretty simple.

First, you must clone the repository [https://github.com/PrestaShop/vagrant](https://github.com/PrestaShop/vagrant).

Second, execute the script `run.sh` (or `run.bat` if you're under Windows) and follow instructions.

> Most of the time, there is a prompt asking for a network interface, remember it's the communication between the virtual machine and our network card.

Reviewer 2: ```suggestion
Most of the time, there is a prompt asking for a network interface, remember it's the communication between the virtual machine and the network card.
```
